securedrop/client.py

```diff
@@ -37,31 +41,76 @@ def make_dict(self):
         }
 
 
+class AESWrapper(object):
+
+    def __init__(self, key):
+        self.bs = AES.block_size
+        self.key = Crypto.Util.Padding.pad(key.encode('ascii'), self.bs)
+
+    def encrypt(self, raw):
+        raw = Crypto.Util.Padding.pad(raw.encode('ascii'), self.bs)
+        iv = get_random_bytes(AES.block_size)
+        cipher = AES.new(self.key, AES.MODE_CBC, iv)
+        bytes_ = base64.b64encode(iv + cipher.encrypt(raw))
+        return bytes_.decode('ascii')
+
+    def decrypt(self, enc):
+        try:
+            data = base64.b64decode(enc)
+            iv = data[:AES.block_size]
+            cipher = AES.new(self.key, AES.MODE_CBC, iv)
+            result = Crypto.Util.Padding.unpad(cipher.decrypt(data[AES.block_size:]), self.bs).decode('utf-8')
+            return result
+        except ValueError:
+            raise ValueError("Decryption was not successful, could not verify input")
+
+
 class ClientData:
-    # todo: encrypt name and contacts
     name: str
     email: str
     contacts: dict
     auth: Authentication
+    email_hash: str
+    enc_name: str
+    enc_contacts: str
 
     def __init__(self, nm=None, em=None, cs=None, password=None, jdict=None):
         if jdict is not None:
-            self.name, self.email, self.contacts, self.auth = \
+            self.enc_name, self.email_hash, self.enc_contacts, self.auth = \
                 jdict["name"], jdict["email"], jdict["contacts"], Authentication(jdict=jdict["auth"])
         else:
             self.name, self.email, self.contacts, self.auth = nm, em, cs, Authentication(password)
+            self.email_hash = hashlib.sha256((self.email.encode())).hexdigest()
+            self.encrypt_name_contacts()
```

Comment:
This line isn't needed since it's called in `make_dict`

Reply:
Yep, you right.